drivers/flash/spi_nor.c

```diff
@@ -555,12 +585,18 @@ static int spi_nor_wrsr(const struct device *dev,
  */
 static int mxicy_rdcr(const struct device *dev)
 {
-	uint16_t cr;
-	enum { CMD_RDCR = 0x15 };
-	int ret = spi_nor_cmd_read(dev, CMD_RDCR, &cr, sizeof(cr));
+	const struct spi_nor_config *cfg = dev->config;
+	uint16_t cr = -ENOSYS;
+
+	if (cfg->mxicy_mx25r_power_mode_exist) {
+		enum {
+			CMD_RDCR = 0x15
```

Comment:
This looks awful (not related to the PR, it looks awful as it was), can you put this with the rest of the defines in the .h or top of this file (where ever they are) and remove this?